### ISSUE TITLE
Static method XLWorkbook.EvaluateExpr doesn't apply default style format (need X decimal Places)

### ISSUE THREAD

**Issue by renenucci** [ClosedXML/ClosedXML#2060]
## Read and complete the full issue template

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature
- [ ] Question

**Did you test against the latest CI build?**
- [ ] Yes
- [X] No

**Version of ClosedXML**

e.g. 0.101

**What is the current behavior?**

The static method doens't have any property to setup Number Format, so I tried setup the Static Default Number format but has no effect at all.

![image](https://user-images.githubusercontent.com/18505538/231470844-e69205a9-9186-4a5f-9f8e-4e96025bf9ef.png)

**What is the expected behavior or new feature?**

Expected the result below:
![image](https://user-images.githubusercontent.com/18505538/231470971-250be178-e4de-47f0-a080-e30688071f45.png)

If I try with "new Worksheet", works fine with this format, but using the static method not.

## Reproducibility

**Code to reproduce problem:**
```c#
XLWorkbook.DefaultStyle.NumberFormat.Format = "0.00000000000";
var t = XLWorkbook.EvaluateExpr("= 14351116316/1000000");
```

**Comment by jahav** [comment]
That is by design.
Static method XLWorkbook.EvaluateExpr is only for evaluating a value. Formating is out of the scope of that method.

Add formula to a cell and use IXLCell.GetFormattedString or use ExcelNumberFormat library.

**Comment by renenucci** [comment]
> That is by design. Static method XLWorkbook.EvaluateExpr is only for evaluating a value. Formating is out of the scope of that method.
> 
> Add formula to a cell and use IXLCell.GetFormattedString or use ExcelNumberFormat library.

I get it, but I'm losing data without any option using the static. The default behavior is rounding up to 6 decimal places.

About 'Add formula to a cell and use IXLCell.GetFormattedString', Can I achieve this without create a entire workbook/sheet object (performance issues)? In my case I have a lot strings that I need to evaluate and round up 11 decimal places. (I Can't loose any decimal place)

**Comment by renenucci** [comment]
@jahav Tried but when I Call GetFormattedString is showing wrong value 
![image](https://user-images.githubusercontent.com/18505538/231496361-399c5fb0-7142-4aa1-b2ee-d2db6556a184.png)

In the saved excel is right:
![image](https://user-images.githubusercontent.com/18505538/231496512-ca7e9cb7-37cc-4d52-a0d1-d4d1b99f71ab.png)


Expected value should be 14351,11631599990 not 14351.11631600000 

It's rounding up and showing after the 11th decimal places.

the formula is '= 14351116316/1000000'

**Comment by jahav** [comment]
Float point numbers are inherently imprecise because they can't represent every value. In this case, the closes representation is 1.43511163159999996423721313477E4 
 (https://www.binaryconvert.com/result_double.html?decimal=049052051053049046049049054051049054048048048048) and it is rightfully rounded to 15 digits a correct 15 digit number because the rest is meaningless.

The actual value 1.43511163159999996423721313477E4  is closer to 1.4351116316 than to 1.43511163159999.

As far as I know, C# doesn't have an API to print number beyond possible precision and I don't think I it should even be printed beyond 15-16 digits (precision of double). It's not a goal of ClosedXML to be complete epsilon precise duplicate of Excel.
